chain/client/src/client.rs

```diff
@@ -1105,15 +1089,13 @@ impl ClientActor {
             block_producer.signer.clone(),
         );
 
-        let ret = self.process_block(ctx, block, Provenance::PRODUCED);
-
-        near_chain::test_utils::display_chain(&mut self.chain);
+        // Update latest known even before sending block out, to prevent race conditions.
+        self.chain.mut_store().save_latest_known(LatestKnown {
+            height: next_height,
+            seen: to_timestamp(Utc::now()),
+        })?;
 
-        assert!(
```

Comment:
This assert should not be removed completely, there were bugs previously causing this assert to fail.
To my best understanding of the logic `process_block` should never fail for a freshly computed block.
Replace with `byzantine_assert!` at least?

Reply:
It can fail if there is time issue (like I had it fail when my NTP switched clock backward) or due to some other external reason.